rs/state_machine_tests/src/lib.rs

```diff
@@ -823,14 +823,12 @@ pub struct StateMachine {
     consensus_time: Arc<PocketConsensusTime>,
     ingress_pool: Arc<RwLock<PocketIngressPool>>,
     ingress_manager: Arc<IngressManager>,
-    pub ingress_filter:
-        tower::buffer::Buffer<IngressFilterService, (ProvisionalWhitelist, SignedIngressContent)>,
+    pub ingress_filter: Arc<Mutex<IngressFilterService>>,
```

Comment:
Why do we need a mutex here and below given that we don't modify the underlying service?

Reply:
Because due to all crazy dependencies the ingress filter needs to be Send+Sync.

Reply:
OK. Could we then please drop the lock right after using the respective service (to avoid potential deadlocks)?

Reply:
You clone the service, so the lock is immediately released